src/libbson/tests/test-json.c

```diff
@@ -3367,6 +3368,318 @@ test_parse_array (void)
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_RELAXED, "{ 'long' : 456 }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_LEGACY, "{ 'long' : 456 }");
+         bson_destroy (b);
+      }
+
+      // BSON type: decimal
+      {
+         bson_decimal128_t decimal128;
+         ASSERT (bson_decimal128_from_string ("1.23", &decimal128));
+
+         bson_t *b = BCON_NEW ("decimal", BCON_DECIMAL128 (&decimal128));
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_CANONICAL, "{ 'decimal' : { '$numberDecimal' : '1.23' } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_RELAXED, "{ 'decimal' : { '$numberDecimal' : '1.23' } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_LEGACY, "{ 'decimal' : { '$numberDecimal' : '1.23' } }");
+         bson_destroy (b);
+      }
+
+      // BSON type: minKey
+      {
+         bson_t *b = BCON_NEW ("minKey", BCON_MINKEY);
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_CANONICAL, "{ 'minKey' : { '$minKey' : 1 } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_RELAXED, "{ 'minKey' : { '$minKey' : 1 } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_LEGACY, "{ 'minKey' : { '$minKey' : 1 } }");
+         bson_destroy (b);
+      }
+
+      // BSON type: maxKey
+      {
+         bson_t *b = BCON_NEW ("maxKey", BCON_MAXKEY);
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_CANONICAL, "{ 'maxKey' : { '$maxKey' : 1 } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_RELAXED, "{ 'maxKey' : { '$maxKey' : 1 } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_LEGACY, "{ 'maxKey' : { '$maxKey' : 1 } }");
+         bson_destroy (b);
+      }
+   }
+
+
+   // Test other cases referenced in the Extended JSON specification that sometimes encode differently.
+   {
+      // Code with scope
+      {
+         bson_t *scope = BCON_NEW ("a", "b");
+
+         bson_t *b = BCON_NEW ("codewscope", BCON_CODEWSCOPE ("code", scope));
+         ASSERT_AS_JSON (
+            b, BSON_JSON_MODE_CANONICAL, "{ 'codewscope' : { '$code' : 'code', '$scope' : { 'a' : 'b' } } }");
+         ASSERT_AS_JSON (
+            b, BSON_JSON_MODE_RELAXED, "{ 'codewscope' : { '$code' : 'code', '$scope' : { 'a' : 'b' } } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_LEGACY, "{ 'codewscope' : { '$code' : 'code', '$scope' : { 'a' : 'b' } } }");
+         bson_destroy (scope);
+         bson_destroy (b);
+      }
+
+      // Double NaN.
+      {
+         bson_t *b = BCON_NEW ("double_nan", BCON_DOUBLE (NAN));
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_CANONICAL, "{ 'double_nan' : { '$numberDouble' : 'NaN' } }");
+         ASSERT_AS_JSON (b, BSON_JSON_MODE_RELAXED, "{ 'double_nan' : { '$numberDouble' : 'NaN' } }");
+         // Legacy encoding is platform dependent. May encode to '[-]nan' or '[-]nan(n-char-sequence)' without quotes.
+         {
+            char *as_legacy = bson_as_legacy_extended_json (b, NULL);
+            char *ptr = as_legacy;
+            ASSERT_STARTSWITH (ptr, "{ \"double_nan\" : ");
+            ptr += strlen ("{ \"double_nan\" : ");
```

Comment:
This seems barely worth mentioning, but it makes me a bit nervous to have two copies of what's really the same prefix string. If they get out of sync the following comparison will be made at the wrong location, which should be detected immediately. Still, it might be a little less error prone to define a const local var for this.

Reply:
Seems reasonable enough. Updated.